sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/TPCDSQueryBenchmark.scala

```diff
@@ -32,76 +33,76 @@ import org.apache.spark.util.Benchmark
-        case UnresolvedRelation(t: TableIdentifier) =>
-          queryRelations.add(t.table)
-        case lp: LogicalPlan =>
-          lp.expressions.foreach { _ foreach {
-            case subquery: SubqueryExpression =>
-              subquery.plan.foreach {
-                case UnresolvedRelation(t: TableIdentifier) =>
-                  queryRelations.add(t.table)
-                case _ =>
-              }
-            case _ =>
+
+    private val tables = Seq("catalog_page", "catalog_returns", "customer", "customer_address",
+      "customer_demographics", "date_dim", "household_demographics", "inventory", "item",
+      "promotion", "store", "store_returns", "catalog_sales", "web_sales", "store_sales",
+      "web_returns", "web_site", "reason", "call_center", "warehouse", "ship_mode", "income_band",
+      "time_dim", "web_page")
+
+    private def setupTables(dataLocation: String): Map[String, Long] = {
+      tables.map { tableName =>
+        spark.read.parquet(s"$dataLocation/$tableName").createOrReplaceTempView(tableName)
+        tableName -> spark.table(tableName).count()
+      }.toMap
+    }
+
+    def run(): Unit = {
+      val tableSizes = setupTables(dataLocation)
+      queries.foreach { name =>
+        val queryString = fileToString(new File(Thread.currentThread().getContextClassLoader
+          .getResource(s"tpcds/$name.sql").getFile))
+
+        // This is an indirect hack to estimate the size of each query's input by traversing the
+        // logical plan and adding up the sizes of all tables that appear in the plan. Note that this
+        // currently doesn't take WITH subqueries into account which might lead to fairly inaccurate
+        // per-row processing time for those cases.
+        val queryRelations = scala.collection.mutable.HashSet[String]()
+        spark.sql(queryString).queryExecution.logical.map {
+          case UnresolvedRelation(t: TableIdentifier) =>
+            queryRelations.add(t.table)
+          case lp: LogicalPlan =>
+            lp.expressions.foreach { _ foreach {
+              case subquery: SubqueryExpression =>
+                subquery.plan.foreach {
+                  case UnresolvedRelation(t: TableIdentifier) =>
+                    queryRelations.add(t.table)
+                  case _ =>
+                }
+              case _ =>
+            }
           }
+          case _ =>
         }
-        case _ =>
-      }
-      val numRows = queryRelations.map(tableSizes.getOrElse(_, 0L)).sum
-      val benchmark = new Benchmark(s"TPCDS Snappy", numRows, 5)
-      benchmark.addCase(name) { i =>
-        spark.sql(queryString).collect()
+        val numRows = queryRelations.map(tableSizes.getOrElse(_, 0L)).sum
+        val benchmark = new Benchmark(s"TPCDS Snappy", numRows, 5)
+        benchmark.addCase(name) { i =>
+          spark.sql(queryString).collect()
+        }
+        logInfo(s"\n\n===== TPCDS QUERY BENCHMARK OUTPUT FOR $name =====\n")
```

Comment:
This change is not directly related to this pr though, I added this log here cuz this change is much trivial. I think this log helps to check which query fails.